### PR TITLE
[AIRFLOW-4163] IntervalCheckOperator supports relative diff

### PR DESCRIPTION
**Pull request by milton0825**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira]
  - https://issues.apache.org/jira/browse/AIRFLOW-4163

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Make `IntervalCheckOperator` supports calculating ratio with relative difference. And also add an option to not ignore zero values in metric.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by milton0825** [comment]
Tests passed on: https://travis-ci.org/milton0825/airflow/builds/511837166

**Comment by milton0825** [comment]
PTAL @feng-tao @ashb

**Comment by Fokko** [comment]
@milton0825 can you rebase, this issue has been fixed on master.

**Comment by feng-tao** [comment]
will take a look.

**Comment by milton0825** [comment]
@Fokko  tests passed~

File: airflow/operators/check_operator.py [hunk 1]
```diff
@@ -248,14 +275,21 @@ def execute(self, context=None):
...
-                ratio = float(max(current[m], reference[m])) / \
-                    min(current[m], reference[m])
-            self.log.info("Ratio for %s: %s \n Ratio threshold : %s", m, ratio, self.metrics_thresholds[m])
-            ratios[m] = ratio
-            test_results[m] = ratio < self.metrics_thresholds[m]
+                ratios[m] = self.ratio_formulas[self.ratio_formula](current[m], reference[m])
+                test_results[m] = ratios[m] < threshold
+
+            self.log.info("Current metric for %s: %s", m, cur)
+            self.log.info("Past metric for %s: %s", m, ref)
```
Comment (feng-tao) [inline airflow/operators/check_operator.py]: do we need that many logging info lines? Let's change it to debug. WDYT?

**Review by feng-tao — APPROVED** [review]
one nit, lgtm

File: airflow/operators/check_operator.py (hunk 1 above)
Comment (milton0825) [reply airflow/operators/check_operator.py]: Correct me if I am wrong. I feel like we only turn on debug mode when there is an Airflow internal issue and we are trying to debug the Airflow internal issue. 

For this case, task failure is usually caused by incorrect user behavior, e.g. wrong SQL (not related to Airflow internal), and we should provide the information users need to debug their issues.

Comment (feng-tao) [reply airflow/operators/check_operator.py]: Ok, then could we make it a single log line(we could add \d in between) which is same as previous case?

Comment (milton0825) [reply airflow/operators/check_operator.py]: fixed.

**Comment by milton0825** [comment]
@feng-tao tests passed on https://travis-ci.org/milton0825/airflow/builds/515481016

**Comment by feng-tao** [comment]
lgtm, thanks @milton0825

**Pull request merged**